tests/scripts/crossgen_comparison.py

```diff
@@ -0,0 +1,372 @@
+    'System.Security.Cryptography.X509Certificates.dll',
+    'System.Security.dll',
+    'System.Security.Permissions.dll',
+    'System.Security.Principal.dll',
+    'System.Security.Principal.Windows.dll',
+    'System.Security.SecureString.dll',
+    'System.ServiceModel.Web.dll',
+    'System.ServiceProcess.dll',
+    'System.Text.Encoding.CodePages.dll',
+    'System.Text.Encoding.dll',
+    'System.Text.Encoding.Extensions.dll',
+    'System.Text.RegularExpressions.dll',
+    'System.Threading.AccessControl.dll',
+    'System.Threading.dll',
+    'System.Threading.Overlapped.dll',
+    'System.Threading.Tasks.Dataflow.dll',
+    'System.Threading.Tasks.dll',
+    'System.Threading.Tasks.Extensions.dll',
+    'System.Threading.Tasks.Parallel.dll',
+    'System.Threading.Thread.dll',
+    'System.Threading.ThreadPool.dll',
+    'System.Threading.Timer.dll',
+    'System.Transactions.dll',
+    'System.Transactions.Local.dll',
+    'System.ValueTuple.dll',
+    'System.Web.dll',
+    'System.Web.HttpUtility.dll',
+    'System.Windows.dll',
+    'System.Xml.dll',
+    'System.Xml.Linq.dll',
+    'System.Xml.ReaderWriter.dll',
+    'System.Xml.Serialization.dll',
+    'System.Xml.XDocument.dll',
+    'System.Xml.XmlDocument.dll',
+    'System.Xml.XmlSerializer.dll',
+    'System.Xml.XPath.dll',
+    'System.Xml.XPath.XDocument.dll',
+    'TraceReloggerLib.dll',
+    'WindowsBase.dll',
+    'xunit.abstractions.dll',
+    'xunit.assert.dll',
+    'xunit.core.dll',
+    'xunit.execution.dotnet.dll',
+    'xunit.performance.core.dll',
+    'xunit.performance.execution.dll',
+    'xunit.performance.metrics.dll',
+    'xunit.runner.utility.dotnet.dll']
+
+class CrossGenRunner:
+    def __init__(self, crossgen_executable_filename, no_logo=True):
+        self.crossgen_executable_filename = crossgen_executable_filename
+        self.no_logo = no_logo
+        self.platform_assemblies_paths_sep = ";"
+
+    def run(self, in_filename, out_filename, platform_assemblies_paths):
+        p = subprocess.Popen(self._build_args(in_filename, out_filename, platform_assemblies_paths), stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = p.communicate()
+        return (p.returncode, stdout.decode(), stderr.decode())
+
+    def _build_args(self, in_filename, out_filename, platform_assemblies_paths):
+        args = []
+        args.append(self.crossgen_executable_filename)
+        if self.no_logo:
+           args.append('/nologo')
+        args.append('/Platform_Assemblies_Paths')
+        args.append(self.platform_assemblies_paths_sep.join(platform_assemblies_paths))
+        args.append('/out')
+        args.append(out_filename)
+        args.append(in_filename)
+        return args
+
+def compute_file_hashsum(filename):
+    algo=hashlib.sha256()
+    block_size_in_bytes = 65536
+    with open(filename, 'rb') as file:
+        while True:
+            block = file.read(block_size_in_bytes)
```

Comment:
Q: Does this read "up to block_size_in_bytes"? Wouldn't want it to fail or skip the data if it was less.

Reply:
Correct. It means at most `block_size_in_bytes` bytes to read.  If the EOF has been reached, `file.read()` will return `None`.